### PR TITLE
Build for 232 EAP 9

### PR DESCRIPTION
**Pull request by vchikoti1998**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: buildSrc/src/main/kotlin/software/aws/toolkits/gradle/intellij/IdeVersions.kt [hunk 1]
```diff
@@ -181,20 +181,20 @@ object IdeVersions {
...
-                    "org.jetbrains.plugins.go:232.6734.9",
+                    "Pythonid:232.8453.116",
+                    "org.jetbrains.plugins.go:232.8453.116",
                     // https://github.com/JetBrains/gradle-intellij-plugin/issues/1056
                     "org.intellij.intelliLang"
                 )
             ),
             rider = RiderProfile(
-                sdkVersion = "2023.2-EAP2-SNAPSHOT",
+                sdkVersion = "2023.2-EAP6-SNAPSHOT",
```
Comment (rli) [inline buildSrc/src/main/kotlin/software/aws/toolkits/gradle/intellij/IdeVersions.kt:191]: EAP7

Comment (vchikoti1998) [reply buildSrc/src/main/kotlin/software/aws/toolkits/gradle/intellij/IdeVersions.kt:191]: yeah, already updated, will put a commit

File: jetbrains-ultimate/src-222-231/software/aws/toolkits/jetbrains/CreateConstraints.kt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains
+
+typealias CreateConstraints = com.intellij.database.dataSource.url.ui.UrlPropertiesPanel
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply jetbrains-ultimate/src-222-231/software/aws/toolkits/jetbrains/CreateConstraints.kt]: resolved

File: jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetLambdaHandlerResolver.kt [hunk 3]
```diff
@@ -1,31 +1,31 @@
...
+ import com.jetbrains.rider.projectView.workspace.isProject
+ import software.aws.toolkits.core.utils.getLogger
+ import software.aws.toolkits.core.utils.warn
+ import software.aws.toolkits.jetbrains.protocol.HandlerExistRequest
+ import software.aws.toolkits.jetbrains.protocol.lambdaPsiModel
+ import software.aws.toolkits.jetbrains.services.lambda.LambdaHandlerResolver
+ import software.aws.toolkits.jetbrains.services.lambda.dotnet.element.RiderLambdaHandlerFakePsiElement
+ import java.nio.file.Path
+
+ class DotNetLambdaHandlerResolver : LambdaHandlerResolver {
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply jetbrains-rider/src/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetLambdaHandlerResolver.kt]: resolved

File: jetbrains-rider/tst/software/aws/toolkits/jetbrains/services/lambda/completion/DotNetHandlerCompletionProviderTest.kt [hunk 4]
```diff
@@ -11,7 +11,7 @@ import software.aws.toolkits.core.lambda.LambdaRuntime
 
 class DotNetHandlerCompletionProviderTest : AwsReuseSolutionTestBase() {
 
-    override fun getSolutionDirectoryName(): String = "SamHelloWorldApp"
+    override fun getSolutionDirectoryName(): String = "testData/solutions/SamHelloWorldApp/SamHelloWorldApp.sln"
```
Comment (manodnyab) [inline jetbrains-rider/tst/software/aws/toolkits/jetbrains/services/lambda/completion/DotNetHandlerCompletionProviderTest.kt:14]: The SamHelloWorldApp directory name will be required for previous versions before 232, correct?

File: jetbrains-rider/tst/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetLambdaHandlerResolverTest.kt [hunk 5]
```diff
@@ -16,7 +16,7 @@ import software.aws.toolkits.jetbrains.services.lambda.Lambda
  */
 class DotNetLambdaHandlerResolverTest : AwsReuseSolutionTestBase() {
 
-    override fun getSolutionDirectoryName(): String = "SamHelloWorldApp"
+    override fun getSolutionDirectoryName(): String = "testData/solutions/SamHelloWorldApp/SamHelloWorldApp.sln"
```
Comment (manodnyab) [inline jetbrains-rider/tst/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetLambdaHandlerResolverTest.kt:19]: same comment as above

File: jetbrains-ultimate/tst/com/goide/vgo/VgoTestUtil.java [hunk 6]
```diff
@@ -63,7 +63,7 @@ private static VgoDependencyImpl dependency(@NotNull String importPath,
                                                 @Nullable VgoDependencyImpl replace) {
         String dirName = dir != null ? dir : String.format("%s@%s", importPath, version);
         String dirPath = FileUtil.join(GOPATH, "pkg", GoConstants.VGO_DIR_NAME, dirName);
-        return new VgoDependencyImpl(importPath, version, null, PathUtil.toSystemIndependentName(FileUtil.toCanonicalPath(dirPath)), replace, false);
+        return new VgoDependencyImpl(importPath, version, null, PathUtil.toSystemIndependentName(FileUtil.toCanonicalPath(dirPath)), replace, false, null, null, null, null);
```
Comment (manodnyab) [inline jetbrains-ultimate/tst/com/goide/vgo/VgoTestUtil.java:66]: Will this be compatible with previous versions?

File: jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/JavaTestUtils.kt [hunk 7]
```diff
@@ -269,7 +269,7 @@ internal fun HeavyJavaCodeInsightTestFixtureRule.setUpMavenProject(): PsiClass {
         // will be re-added in next EAP
         // https://github.com/JetBrains/intellij-community/commit/0488a28a89b661c2cbf0706060bee2f7bd367237
         // projectsManager.waitForResolvingCompletion()
-        projectsManager.scheduleImportInTests(poms)
+        // projectsManager.scheduleImportInTests(poms)
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply jetbrains-core/tst/software/aws/toolkits/jetbrains/utils/JavaTestUtils.kt]: resoolved

File: jetbrains-rider/src-232+/software/aws/toolkits/jetbrains/utils/OpenSolutionDirName.kt [hunk 8]
```diff
@@ -0,0 +1,6 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.utils
+
+const val OPEN_SOLUTION_DIR_NAME: String = "testData/solutions/SamHelloWorldApp/SamHelloWorldApp.sln"
```
Comment (manodnyab) [inline jetbrains-rider/src-232+/software/aws/toolkits/jetbrains/utils/OpenSolutionDirName.kt:6]: This const along with the function openSolutionFile could be in a single util file?

File: jetbrains-rider/tst/base/RiderTestFrameworkUtils.kt [hunk 9]
```diff
@@ -60,9 +58,14 @@ val msBuild by lazy {
...
+//    VfsRootAccess.allowRootAccess(
+//        ApplicationManager.getApplication(),
+//        dotNetSdk,
+//        File(TestEnvironmentInstance.dotnetCoreCliPath).parentFile.absolutePath
+//    )
+
     VfsRootAccess.allowRootAccess(
         ApplicationManager.getApplication(),
-        dotNetSdk,
-        File(PrepareTestEnvironment.dotnetCoreCliPath).parentFile.absolutePath
```
Comment (manodnyab) [inline jetbrains-rider/tst/base/RiderTestFrameworkUtils.kt:66]: Is this file required in the versions prior to 232?

File: jetbrains-ultimate/tst/com/goide/vgo/VgoTestUtil.java (hunk 6 above)
Comment (vchikoti1998) [reply jetbrains-ultimate/tst/com/goide/vgo/VgoTestUtil.java]: resloved

File: buildSrc/src/main/kotlin/software/aws/toolkits/gradle/intellij/IdeVersions.kt [hunk 10]
```diff
@@ -181,20 +181,20 @@ object IdeVersions {
...
             rider = RiderProfile(
-                sdkVersion = "2023.2-EAP2-SNAPSHOT",
+                sdkVersion = "2023.2-EAP8-SNAPSHOT",
                 plugins = commonPlugins223 + listOf(
                     "rider-plugins-appender" // Workaround for https://youtrack.jetbrains.com/issue/IDEA-179607
                 ),
                 netFrameworkTarget = "net472",
-                rdGenVersion = "2023.2.1",
-                nugetVersion = "2023.2.0-eap02"
+                rdGenVersion = "2023.2.3", // https://central.sonatype.com/artifact/com.jetbrains.rd/rd-gen/2023.2.2/versions
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply buildSrc/src/main/kotlin/software/aws/toolkits/gradle/intellij/IdeVersions.kt]: resolved

File: jetbrains-rider/tst-231+/software/aws/toolkits/jetbrains/services/lambda/compat/CachedImageIcon.kt [hunk 11]
```diff
@@ -3,4 +3,5 @@
 
 package software.aws.toolkits.jetbrains.services.lambda.compat
 
-typealias CachedImageIcon = com.intellij.openapi.util.CachedImageIcon
+// typealias CachedImageIcon = com.intellij.openapi.util.CachedImageIcon
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply jetbrains-rider/tst-231+/software/aws/toolkits/jetbrains/services/lambda/compat/CachedImageIcon.kt:6]: typo mistake, resolved

File: jetbrains-rider/tst/base/setUpCustomToolset.kt [hunk 12]
```diff
@@ -1,10 +1,10 @@
...
+// // Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// // SPDX-License-Identifier: Apache-2.0
+//
+// package base
+//
+// import com.jetbrains.rider.test.scriptingApi.setUpCustomToolset
+//
+// // FIX_WHEN_MIN_IS_221: signature changed in 221
+// fun setUpCustomToolset(path: String, host: com.jetbrains.rdclient.protocol.IProtocolHost) =
+//    setUpCustomToolset(path, host)
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply jetbrains-rider/tst/base/setUpCustomToolset.kt]: resolved

File: jetbrains-rider/tst/base/RiderTestFrameworkUtils.kt [hunk 13]
```diff
@@ -60,9 +58,14 @@ val msBuild by lazy {
 fun allowCustomDotnetRoots() {
     // Rider Test Framework miss VFS root access for the case when running tests on local environment with custom SDK path
     // This should be fixed on Rider Test Framework level. Workaround it until related ticket RIDER-47995 is fixed.
+//    VfsRootAccess.allowRootAccess(
+//        ApplicationManager.getApplication(),
+//        dotNetSdk,
+//        File(TestEnvironmentInstance.dotnetCoreCliPath).parentFile.absolutePath
+//    )
```
[the remark this answers is not in the document]
Comment (vchikoti1998) [reply jetbrains-rider/tst/base/RiderTestFrameworkUtils.kt:65]: removed it

File: jetbrains-rider/tst/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetLambdaHandlerResolverTest.kt (hunk 5 above)
Comment (vchikoti1998) [reply jetbrains-rider/tst/software/aws/toolkits/jetbrains/services/lambda/dotnet/DotNetLambdaHandlerResolverTest.kt]: Resolved

File: jetbrains-rider/src-232+/software/aws/toolkits/jetbrains/utils/OpenSolutionDirName.kt (hunk 8 above)
Comment (vchikoti1998) [reply jetbrains-rider/src-232+/software/aws/toolkits/jetbrains/utils/OpenSolutionDirName.kt]: Resolved

**Comment by vchikoti1998** [comment]
Closing, worked on another branch
